### ISSUE TITLE
Uncheck a preselected option unchecks the last option with showCheckbox activated

### ISSUE THREAD

**Issue by McBenny** [srigar/multiselect-react-dropdown#98]
When I have a Multiselect that has pre-selected values and displays checkboxes, if I open the dropdown by clicking on the placeholder, and then click on one of the pre-selected values to "uncheck" it (click on the checkbox itself or the label), the actual option being deselected is not the one I clicked on but the last of the pre-selected values.

Clicking on the cross of the value without opening the dropdown works fine though.

In the following example, there are 3 options, and all 3 are pre-selected. If I open the dropdown and click on the 2nd option to unselect it, the 3rd option (last one) is unselected.
```
<Multiselect
    options={
        [
            {id: 1, name: 'option 1'},
            {id: 2, name: 'option 2'},
            {id: 3, name: 'option 3'}
        ]
    }
    displayValue="name"
    showCheckbox
    placeholder="My placeholder"
    selectedValues={
        [
            {id: 1, name: 'option 1'},
            {id: 2, name: 'option 2'},
            {id: 3, name: 'option 3'}
        ]
    }
    onSelect={
        (selectedList, selectedItem) => console.log('select', selectedList, selectedItem)
    }
    onRemove={
        (selectedList, selectedItem) => console.log('remove', selectedList, selectedItem)
    }
/>
```
The `selectedItem` is the actual selectedItem but the selectedList is updated with removing the last selected Item instead of the real elected item.

Here is the console on remove event, the list is updated by removing the 3rd option, but the selectedItem is the 2nd.
![multi-select-bug](https://user-images.githubusercontent.com/12895368/108006720-5888ff80-7050-11eb-891b-750bd33e8d24.png)

I'm using node v12.18.2, and multiselect-react-dropdown@1.6.7 with React@16.13.1

**Comment by srigar** [comment]
Changes available in v1.6.11. Pls feel free to reopen if still issue exist.

**Comment by McBenny** [comment]
I can confirm the issue is fixed for me with version 1.6.11.
Thanks mate, that's quick and effective work!
